PubNub/Core/PubNub+Objects.m

```diff
@@ -514,284 +438,487 @@ - (void)sendFetchChannelMembersRequestUsingBuilderParameters:(NSDictionary *)par
+
+        [self callBlock:block status:NO withResult:result.result andStatus:result.status];
+    };
+
+    [self performRequest:userRequest withParser:responseParser completion:handler];
 }
 
 #pragma mark - Channel metadata object
 
-- (void)setChannelMetadataWithRequest:(PNSetChannelMetadataRequest *)request
-                           completion:(nullable PNSetChannelMetadataCompletionBlock)block {
-    
-    __weak __typeof(self) weakSelf = self;
-    
-    [self performRequest:request withCompletion:^(PNSetChannelMetadataStatus *status) {
-        if (block && status.isError) {
-            status.retryBlock = ^{
-                [weakSelf setChannelMetadataWithRequest:request completion:block];
+- (void)setChannelMetadataWithRequest:(PNSetChannelMetadataRequest *)userRequest
+                           completion:(PNSetChannelMetadataCompletionBlock)handleBlock {
+    PNOperationDataParser *responseParser = [self parserWithStatus:[PNSetChannelMetadataStatus class]];
+    PNSetChannelMetadataCompletionBlock block = [handleBlock copy];
+    PNParsedRequestCompletionBlock handler; 
+
+#ifndef PUBNUB_DISABLE_LOGGER
+#endif // PUBNUB_DISABLE_LOGGER
+                           
+    PNWeakify(self);
+    handler = ^(PNTransportRequest *request, id<PNTransportResponse> response, __unused NSURL *location,
+                PNOperationDataParseResult<PNSetChannelMetadataStatus *, PNSetChannelMetadataStatus *> *result) {
+        PNStrongify(self);
+
+        if (result.status.isError) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+            result.status.retryBlock = ^{
+                [self setChannelMetadataWithRequest:userRequest completion:block];
             };
+#pragma clang diagnostic pop
         }
-        
-        [weakSelf callBlock:block status:YES withResult:nil andStatus:status];
-    }];
+
+        [self callBlock:block status:YES withResult:nil andStatus:result.status];
+    };
+
+    [self performRequest:userRequest withParser:responseParser completion:handler];
 }
 
-- (void)removeChannelMetadataWithRequest:(PNRemoveChannelMetadataRequest *)request
-                              completion:(nullable PNRemoveChannelMetadataCompletionBlock)block {
-    
-    __weak __typeof(self) weakSelf = self;
-    
-    [self performRequest:request withCompletion:^(PNAcknowledgmentStatus *status) {
-        if (block && status.isError) {
-            status.retryBlock = ^{
-                [weakSelf removeChannelMetadataWithRequest:request completion:block];
+- (void)removeChannelMetadataWithRequest:(PNRemoveChannelMetadataRequest *)userRequest
+                              completion:(PNRemoveChannelMetadataCompletionBlock)handleBlock {
+    PNOperationDataParser *responseParser = [self parserWithStatus:[PNAcknowledgmentStatus class]];
+    PNRemoveChannelMetadataCompletionBlock block = [handleBlock copy];
+    PNParsedRequestCompletionBlock handler; 
+
+#ifndef PUBNUB_DISABLE_LOGGER
```

Comment:
Here too, perhaps the best way would be to find all these empty occurrences and delete them

Reply:
All Object API files has this as a placeholder. Need to sneak pick into another language which prints something and repeat the message here :)